eogrow/pipelines/split_grid.py

```diff
@@ -0,0 +1,184 @@
+"""Implements a pipeline that creates a finer grid and splits EOPatches accordingly."""
+import logging
+from collections import defaultdict
+from typing import Dict, List, Literal, Tuple, Union
+
+import fs
+import geopandas as gpd
+from pydantic import Field
+
+from eolearn.core import EONode, EOWorkflow, FeatureType, LoadTask, OverwritePermission, SaveTask
+from sentinelhub import BBox
+
+from ..core.area.batch import BatchAreaManager
+from ..core.area.utm import UtmZoneAreaManager
+from ..core.pipeline import Pipeline
+from ..tasks.spatial import SpatialSliceTask
+from ..utils.fs import LocalFile
+from ..utils.grid import split_bbox
+from ..utils.types import Feature, FeatureSpec
+
+LOGGER = logging.getLogger(__name__)
+
+NamedBBox = Tuple[str, BBox]
+
+
+class SplitGridPipeline(Pipeline):
+    """Pipeline that creates a finer grid and splits EOPatches accordingly.
+
+    The new grid is output as a geopackage file, which can be used with a `CustomAreaManager`.
+    """
+
+    class Schema(Pipeline.Schema):
+        input_folder_key: str = Field(
+            description="A storage manager key pointing to the folder from where data will be loaded."
+        )
+        eopatch_output_folder_key: str = Field(
+            description="A storage manager key pointing to the folder to where data will be saved."
+        )
+        grid_output_folder_key: str = Field(
+            description="A storage manager key of where to save the resulting split grid."
+        )
+        subsplit_grid_filename: str = Field(
+            description="Filename of new grid, which can be used in `CustomAreaManager`.", regex=r"^.+\.gpkg$"
+        )
+        features: List[Feature] = Field(description="Which features will be loaded and joined.")
+        raise_misaligned: bool = Field(
+            True,
+            description=(
+                "If True this will raise an error if splitting or joining any spatial raster EOPatch feature would "
+                "cause a misalignment. If False, misalignment issues will be ignored."
```

Comment:
probably can be simplified a bit, perhaps:

"Raise an error if spatially splitting the EOPatch produces misalignments."

I also removed the "or joining" part since this is probably split only?

Reply:
done